connection_manager.go

```diff
@@ -153,6 +154,26 @@ func (n *connectionManager) Run() {
 	}
 }
 
+// Check if peer's certificate is not expired or invalid.
+func (n *connectionManager) checkToDisconnect(now time.Time, hostinfo *HostInfo) (bool, *cert.NebulaCertificate) {
+	if !n.intf.disconnectInvalid {
+		return false, nil
+	}
+
+	if hostinfo == nil {
```

Comment:
Since `hostinfo` will have a nil check before this function, we can remove this re-check.